lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,192 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class SumUpGateway < Gateway
+      self.live_url = 'https://api.sumup.com/v0.1/'
+
+      self.supported_countries = %w(AT BE BG BR CH CL CO CY CZ DE DK EE ES FI FR
+                                    GB GR HR HU IE IT LT LU LV MT NL NO PL PT RO
+                                    SE SI SK US)
+      self.currencies_with_three_decimal_places = %w(EUR BGN BRL CHF CZK DKK GBP
+                                                     HUF NOK PLN SEK USD)
+      self.default_currency = 'USD'
+
+      self.homepage_url = 'https://www.sumup.com/'
+      self.display_name = 'SumUp'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        multiple_invalid_parameters: 'MULTIPLE_INVALID_PARAMETERS'
+      }
+
+      PAYMENT_TYPES = %(card boleto ideal sofort bancontact)
+
+      def initialize(options = {})
+        requires!(options, :token, :pay_to_email)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        MultiResponse.run do |r|
+          r.process { upsert_checkout(money, payment, options) }
+          r.process { complete_checkout(r.params['id'], payment, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Bearer )\w+), '\1[FILTERED]').
+          gsub(%r(("pay_to_email\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def upsert_checkout(money, payment, options = {})
+        if options[:checkout_id].present?
+          find_checkout(options[:checkout_id])
```

Comment:
❓ Question:

What is the point of finding a checkout to extract it's ID if you already have the ID on the `options[:checkout_id]` ?

Reply:
Good catch, you're right